System/Process/Common.hs

```diff
@@ -270,8 +272,11 @@ mbPipe CreatePipe pfd  mode = fmap Just (pfdToHandle pfd mode)
 mbPipe _std      _pfd _mode = return Nothing
 
 pfdToHandle :: Ptr FD -> IOMode -> IO Handle
-pfdToHandle pfd mode = do
-  fd <- peek pfd
+pfdToHandle pfd mode =
+  rawFdToHandle mode =<< peek pfd
+
+rawFdToHandle :: IOMode -> FD -> IO Handle
+rawFdToHandle mode fd = do
```

Comment:
It's a bit strange that `rawFdToHandle` has opposite argument order to `pfdToHandle`.